aries_cloudagent/conductor.py

```diff
@@ -277,9 +284,40 @@ def __init__(self, context_builder: ContextBuilder) -> None:
             parsed_msg, delivery, connection, self.outbound_message_router
         )
         if socket:
+            # if a reply mode is present then a response is allowed over this connection
+            # wait till dispatch is complete,
+            # then check to see if any queued responses can be sent.
+            if socket.reply_mode:
+                await complete
+                await self.queue_processing(socket)
+
+            # close a single_response socket if no answer given by now.
             complete.add_done_callback(lambda fut: socket.dispatch_complete())
         return complete
 
+    async def queue_processing(self, socket):
+        """
+        Interact with undelivered queue to find applicable messages.
+
+        Args:
+            socket: The incoming socket connection
+        """
+
```

Comment:
I think this method should check whether the socket is complete first (has `reply_mode` and is not `closed`) rather than checking in `inbound_message_router`.

Reply:
I think this suggested change depends on where we call `queue_processing()` per the previous comment. I'll wait for that resolution before resolving this.

Reply:
Interestingly, `socket.select_outgoing()` performs both of these checks, so it may not be needed in either place.

Reply:
I still think it would be best to check the state of the socket before going through the items in the queue, as most of the time we're using a single-response HTTP transport

Reply:
I had placed it above this method for the same reason, but maybe that's not the right place.